src/content/marina/marinaBroker.ts

```diff
@@ -326,6 +328,102 @@ export default class MarinaBroker extends Broker {
           return successMsg();
         }
 
+        case Marina.prototype.broadcastTransaction.name: {
+          this.checkHostnameAuthorization(state);
+          const network = selectNetwork(state);
+          const [signedTxHex] = params as [string];
+
+          // broadcast tx
+          const txid = await broadcastTx(selectEsploraURL(state), signedTxHex);
+          if (!txid) throw new Error('something went wrong with the tx broadcasting');
+
+          // will need tx to lock selected utxos and credit unconfirmed utxos
+          const tx = Transaction.fromHex(signedTxHex);
+
+          // lock selected utxos used in this transaction:
+          // - get all coins from marina
+          // - iterate over all transaction inputs and check if is a coin of ours
+          // - if any, lock selected utxos
+
+          // get all coins from marina
+          const coins = selectUtxos(MainAccountID)(state);
+
+          // array to store all utxos selected in this transaction
+          const selectedUtxos: UnblindedOutput[] = [];
+
+          // find all inputs that are a coin of ours
+          tx.ins.forEach((_in) => {
+            const { hash, index } = _in; // txid and vout
+            const txid = hash.reverse().toString('hex');
```

Comment:
nit:To avoid weird errors later, I'm used to combine `Array.slice()` with reverse func. `hash.slice(). reverse()` ensures that hash variable is not mutated.

Reply:
Fixed by 8d2c329